### PR TITLE
anityachecker: Add option to get stable version

### LINKED ISSUE #121: Anitya checker: ignore pre-releases
Anitya allows filtering out pre-releases, which would make the results of the checker more accurate: https://github.com/fedora-infra/anitya/issues/898

### PR DESCRIPTION
**Pull request by gasinvein**
Fixes #121

### CODE REVIEW

File: src/checkers/anityachecker.py [hunk 1]
```diff
@@ -22,15 +22,17 @@ def check(self, external_data):
...
-        response.raise_for_status()
-        result = response.json()
+        query = {"project_id": external_data.checker_data.get("project-id")}
+        with requests.get(versions_url, params=query) as response:
+            response.raise_for_status()
+            result = response.json()
 
-        latest_version = result["latest_version"]
+        if stable_only:
+            latest_version = result["stable_versions"][0]
```
Comment (gasinvein) [inline src/checkers/anityachecker.py:33]: @Zlopez Is it safe to assume that `stable_versions` list is always sorted and we can use the first item on it as latest stable version?

**Comment by gasinvein** [comment]
Ouch! Looks like `stable_versions` may contain version with prefix not stripped.
<details>
  <summary>This is how it looks like:</summary>

```json
{
  "latest_version": "2020.8",
  "versions": [
    "2020.8",
    "2020.7",
    "2020.6",
    "2020.5",
    "2020.4",
    "2020.3",
    "2020.2",
    "2020.1",
    "2019.6",
    "2019.5",
    "2019.4",
    "2019.3",
    "2019.2",
    "2019.1",
    "2018.9",
    "2018.8",
    "2018.7",
    "2018.6",
    "2018.5",
    "2018.4",
    "2018.3",
    "2018.2",
    "2018.1",
    "2017.15",
    "2017.14",
    "2017.13",
    "2017.12",
    "2017.11",
    "2017.10",
    "2017.9",
    "2017.8",
    "2017.7",
    "2017.6",
    "2017.5",
    "2017.4",
    "2017.3",
    "2017.2",
    "2017.1",
    "2016.15",
    "2016.14",
    "2016.14",
    "2016.13",
    "2016.12",
    "2016.12",
    "2016.11",
    "2016.10",
    "2016.9",
    "2016.8",
    "2016.7",
    "2016.6",
    "2016.5",
    "2016.4",
    "2016.3",
    "2016.2",
    "2016.1",
    "2015.11",
    "2015.10",
    "2015.9",
    "2015.8",
    "2015.7",
    "2015.6",
    "2015.5",
    "2015.4",
    "2015.3",
    "2014.4",
    "untagged-e3a32204dfe69a97a7ad"
  ],
  "stable_versions": [
    "v2020.8",
    "v2020.7",
    "v2020.6",
    "v2020.5",
    "v2020.4",
    "v2020.3",
    "v2020.2",
    "v2020.1",
    "v2019.6",
    "v2019.5",
    "v2019.4",
    "v2019.3",
    "v2019.2",
    "2019.1",
    "2018.9",
    "2018.8",
    "2018.7",
    "2018.6",
    "2018.5",
    "2018.4",
    "2018.3",
    "2018.2",
    "2018.1",
    "2017.15",
    "2017.14",
    "2017.13",
    "2017.12",
    "2017.11",
    "2017.10",
    "2017.9",
    "2017.8",
    "2017.7",
    "2017.6",
    "2017.5",
    "2017.4",
    "2017.3",
    "2017.2",
    "2017.1",
    "2016.15",
    "v2016.14",
    "2016.14",
    "v2016.13",
    "v2016.12",
    "2016.12",
    "2016.11",
    "2016.10",
    "v2016.9",
    "v2016.8",
    "v2016.7",
    "v2016.6",
    "2016.5",
    "v2016.4",
    "v2016.3",
    "v2016.2",
    "v2016.1",
    "v2015.11",
    "v2015.10",
    "v2015.9",
    "v2015.8",
    "v2015.7",
    "v2015.6",
    "v2015.5",
    "v2015.4",
    "v2015.3",
    "2014.4",
    "untagged-e3a32204dfe69a97a7ad"
  ]
}
```

<details>

File: src/checkers/anityachecker.py (hunk 1 above)
Comment (Zlopez) [reply src/checkers/anityachecker.py:33]: The stable versions should be always sorted.

**Comment by Zlopez** [comment]
The prefix shouldn't be there, this is a bug I need to fix.

**Comment by Zlopez** [comment]
I looked at the code and found the issue, will fix it when I find some spare time.

**Comment by gasinvein** [comment]
Thanks. Should we wait for a fix on anitya side or merge this feature anyway? I guess it'll work fine for projects without version prefix.

**Comment by Zlopez** [comment]
You can merge it, the sorting should work even when prefix is involved. I will try to fix it soon.

**Comment by gasinvein** [comment]
A side question - how does Anitya distinguish stable versions? Is it only by version pattern match? Asking because some projects seem to have no pre-release markers, namely [flatpak](https://release-monitoring.org/project/6377/), despite being hosted on github.

**Comment by Zlopez** [comment]
It depends on chosen versioning scheme and you can also add your own patterns that can be used to recognize them.

**Comment by gasinvein** [comment]
But it doesn't query github releases api to determine if it's a pre-release?

**Comment by Zlopez** [comment]
I didn't even know the API provides this information.

**Comment by gasinvein** [comment]
Well, in `https://api.github.com/repos/owner/repo/releases` there is a boolean `prerelease`.

**Comment by Zlopez** [comment]
> Well, in `https://api.github.com/repos/owner/repo/releases` there is a boolean `prerelease`.

We are using both tags and releases and I know plenty of projects aren't tagging them correctly. But I can look if there could be used in Anitya.

**Comment by Zlopez** [comment]
Anitya 1.1.1 is deployed with the fix for API https://github.com/fedora-infra/anitya/releases/tag/1.1.1 Enjoy!

**Comment by gasinvein** [comment]
Great, thanks.

**Pull request merged**